### PR TITLE
feat. facility cover image upload

### LINKED ISSUE #2131: Facility Images
Add a field for adding a Facility Image. There are existing helpers for Uploading Images. 

> Backend changes required

### LINKED ISSUE #2171: Facility Images
There needs to be an option to add an Image to a Facility. 

Backend Changes required. Should make use of existing helper functions for file/image upload

### LINKED ISSUE #3527: Facility card is not symmetrically rounded
image

### PR DESCRIPTION
**Pull request by rithviknishad**
Closes #2171 
Closes #2131 
Fixes #3527 

#### 1. Shows the image on the facilities list page

image

#### 1.5 Shows the image on the facility home page

image

image

#### 2. Edit a facility cover photo

image

#### 3. Or upload a new cover photo

image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by rithviknishad** [comment]
### What's new?

Adds `ImageWithFallback` component and utilised for facility cover image when an image cannot be retrieved.

image

image

image

image

**Comment by nihal467** [comment]
@rithviknishad i can see a github-code-scanning comment, can you check it and let me know, the PR is good for testing and its been approved in the EOD meeting

**Comment by nihal467** [comment]
@rithviknishad 
image

- [ ] the upload page is not responsive

image

- [ ] even though i can upload and change exisiting cover with district admin id, not able to delete the facility cover page

image

- [ ] when i click on the save button, there is no action been take place nor iam receiving any notification

image

there is no cover page ratio is been mentioned in the page, if its been mentioned, it would be good for a user to resize the image before uploading it to the platform.

image

- [ ] a option to crop, reposition the view of the uploaded photo would be great, before working on this, discuss it on the EOD call,

**Comment by rithviknishad** [comment]
- [x] The Image Upload modal is made responsive
image

- [x] The backend PR for delete image permission is done. Waiting for https://github.com/coronasafe/care/pull/1071 to be merged

- [x] Show notification on save and also on delete.
image

**Comment by nihal467** [comment]
test approved

File: src/Components/Facility/CoverImageEditModal.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      <div className="h-screen w-full absolute flex items-center justify-center bg-modal">
+        <form className="m-4 bg-white rounded-xl w-11/12 max-w-3xl min-h-[24rem] max-h-screen overflow-auto flex flex-col shadow">
+          <div className="px-6 py-6 flex flex-col bg-gray-300">
+            <span className="text-xl font-medium">Edit Cover Photo</span>
+            <span className="mt-1 text-gray-700">{facility.name}</span>
+          </div>
+          <div className="flex-1 flex m-8 rounded-lg items-center justify-center">
+            {preview || facility.read_cover_image_url ? (
+              <img
+                src={imgSrc}
```
[the remark this answers is not in the document]
Comment (khavinshankar) [reply src/Components/Facility/CoverImageEditModal.tsx:108]: @rithviknishad try validating the url (`imgSrc`), something like
```js
src={ isValidURL(imgSrc) ? imgSrc : defaultURL }
```

**Review by khavinshankar — APPROVED** [review]
LGTM

**Pull request merged**